RELEASES.md

```diff
@@ -1,3 +1,184 @@
+Version 1.16.0 (2017-03-16)
+===========================
+
+Language
+--------
+
+* Lifetimes in statics ad consts default to `'static`. [RFC 1623]
+* [The compiler's `dead_code` lint now accounts for type aliases][38051].
+* [Uninhabitable enums (those without any variants) no longer permit wildcard
+  match patterns][38069]
+* [Clean up semantics of `self` in an import list][38313]
+* [`Self` may appear in `impl` headers][38920]
+* [`Self` may appear in struct expressions][39282]
+
+Compiler
+--------
+
+* [`rustc` now supports `--emit=metadata`, which causes rustc to emit
+  a `.rmeta` file containing only crate metadata][38571]. This can be
+  used by tools like the Rust Language Service to perform
+  metadata-only builds.
+* [Fix `transmute::<T, U>` where `T` requires a bigger alignment than
+  `U`][38670]
+* [rustc: use -Xlinker when specifying an rpath with ',' in it][38798]
+* [Candidates for type suggestions draw from more sources][38927]
```

Comment:
This is an underestimation :)
It's not only type suggestions, Levenshtein based typo suggestions now work in most places, while previously they worked only for fields and *sometimes* for local variables. Together the overhaul of "no resolution"/"unexpected resolution" errors (https://github.com/rust-lang/rust/pull/38154) they result in large and systematic improvement in resolution diagnostics.